types/consensus_data.go

```diff
@@ -190,36 +183,54 @@ func (ci *ConsensusData) GetAttestationData() (*phase0.AttestationData, error) {
 	return ret, nil
 }
 
-func (ci *ConsensusData) GetBellatrixBlockData() (*bellatrix.BeaconBlock, error) {
-	ret := &bellatrix.BeaconBlock{}
-	if err := ret.UnmarshalSSZ(ci.DataSSZ); err != nil {
-		return nil, errors.Wrap(err, "could not unmarshal ssz")
-	}
-	return ret, nil
-}
-
-func (ci *ConsensusData) GetBellatrixBlindedBlockData() (*bellatrix2.BlindedBeaconBlock, error) {
-	ret := &bellatrix2.BlindedBeaconBlock{}
-	if err := ret.UnmarshalSSZ(ci.DataSSZ); err != nil {
-		return nil, errors.Wrap(err, "could not unmarshal ssz")
-	}
-	return ret, nil
-}
-
-func (ci *ConsensusData) GetCapellaBlockData() (*capella.BeaconBlock, error) {
-	ret := &capella.BeaconBlock{}
-	if err := ret.UnmarshalSSZ(ci.DataSSZ); err != nil {
-		return nil, errors.Wrap(err, "could not unmarshal ssz")
+func (ci *ConsensusData) GetBlockData() (*spec.VersionedBeaconBlock, ssz.HashRoot, error) {
+	switch ci.Version {
```

Comment:
why support these old versions? adds redundant tests and complexity

Reply:
you wrote:
> all consensus data tests need to cover all different versions not just one

Anyhow, @lior-blox asked to split the old #171 PR up to chunks and do it correctly.

When we anticipate a new fork there must be a time when our code should support at least 2 versions so it will automatically adjust to the new version. Currently this is hard to do...

The final goal when all those PRs are merged is to be able to easily add a new fork support while supporting at least one old fork. In practice it will be able to support more than one old fork.

Reply:
If you look at the way the `attestantio` lib implemented `VersionedBeaconBlock` you will see that Oleg's implementation is very much in line with the lib's intentions.

I personally think that the lib's implementation is bad... why do we need all the switch cases? I can think of a smarter less wasteful way utilizing interfaces that won't make us add a new field for every fork...
But given that this is the situation the current code is ok imo

However, since you asked and I agree that this is wasteful, I'll ask @olegshmuelov to have a default case for the old versions that will give out an error...
Otherwise, as new forks will come out this function will become even longer.

Reply:
Removed all versions besides Bellatrix.
Upcoming split PR will include Capella support

Reply:
capella is still there.. if we add a new version (cappella) we need tests for it. If you want you can add it in another PR

Reply:
The plan is definitely to add it in another PR.
We are currently in the process of breaking up the original PR up in order to meet the standard of "reviewable PRs"

Reply:
@alonmuroch

you sure there is still Capella support here?
@olegshmuelov and I can't find it...

I think you may have an older version of the PR in your head before Oleg cleared it up

Reply:
I see it's gone, great.